### PR TITLE
separate session for ML_Utils

### PR DESCRIPTION
**Pull request by lsf37**
Rename `ml_helpers` to `ML_Utils` and make it a separate session, because it has no other Lib dependencies.

Part of #543.

### CODE REVIEW

File: lib/ML_Utils/ROOT [hunk 1]
```diff
@@ -0,0 +1,18 @@
+(*
+ * Copyright 2023, Proofcraft Pty Ltd
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ *)
+
+chapter Lib
+
+session ML_Utils (lib) = Word_Lib +
```
Comment (lsf37) [inline lib/ML_Utils/ROOT:9]: This should actually be just HOL.

File: lib/Apply_Trace.thy [hunk 2]
```diff
@@ -8,8 +8,7 @@
  * Provides an alternate refinement function which takes an additional stateful journaling operation. *)
 theory Apply_Trace
 imports
-  Main
-  "MLUtils"
+  ML_Utils.MLUtils
```
Comment (Xaphiosis) [inline lib/Apply_Trace.thy:11]: Do we not want these to be named consistently? `ML_Utils` vs `MLUtils` on the same line is a bit blah

Comment (Xaphiosis) [reply lib/Apply_Trace.thy:11]: Other than that, looks good to me.

**Review by Xaphiosis — APPROVED** [review]

File: lib/Apply_Trace.thy (hunk 2 above)
Comment (lsf37) [reply lib/Apply_Trace.thy:11]: Didn't want to introduce too much noise, but you do have a point. I'll add that rename.

**Pull request merged**